cli/src/pcluster/templates/cluster_stack.py

```diff
@@ -84,12 +85,19 @@ def __init__(
         stack_name: str,
         cluster_config: SlurmClusterConfig,
         bucket: S3Bucket,
+        log_group_name=None,
         **kwargs,
     ) -> None:
         super().__init__(scope, construct_id, **kwargs)
         self._stack_name = stack_name
         self.config = cluster_config
         self.bucket = bucket
+        if self.config.is_cw_logging_enabled:
+            if log_group_name:
```

Comment:
let's add a comment here describing that this logic is to manage the stack update. In future we can have a function "is_update" to be reused everywhere.